draft/XXXX-orm/README.md

```diff
@@ -0,0 +1,103 @@
+---
+RFC: XXXX
+Author: Jacob Nesbitt
+Status: draft
+Created: 2020-07-28
+Last-Modified: 2020-07-28
+---
+
+# RFC XXXX: Multinet ORM
+
+This RFC addresses the issue of organizing database operations into classes which interact with each other. This aims to reduce the complexity, while improving the clarity and extendability of our database operations.
+
+## Motivation
+
+Currently, our database operations are structured in a purely functional way. That is, all database operations are done through simple functions, with no use of classes. This becomes problematic for readability, as our current `db.py` file has grown to almost 600 lines long. This presents a risk, as lack of code readability/clarity likely leads to more mistakes/bugs being introduced.
+
+Furthermore, due to the flat use of stateless functions, performance becomes an issue, as the same few functions are called repeatedly by multiple functions, as they share no state. The temporary solution to this was caching, but by introducing an ORM, much of the need for this caching will go away, as this state can be preserved in a class instance.
+
+## Proposals
+
+### Classes
+The following subsections describe the classes that make up the Multinet ORM. Type annotations are included to more accurately describe the connections between the classes. If a type annotation is left out, it is either due to uncertainty, or because that method returns nothing useful.
+
+
+##### User
+```
+# Previously used the term "cookie"
+def get_session() -> str: ...
+def set_session(): ...
+def delete_session(): ...
+
+# Serializes the object to JSON
+def serialize() -> Dict: ...
```

Comment:
Presumably, all the ORM classes should have a `serialize()` method, right? That would let us construct different types of JSON responses to various levels of API endpoints, in a more comprehensive way than we do now.

Reply:
I suppose, but there's some open questions regarding that. Like, does the `serialize` method dump everything into JSON, or is it selectively outputting information? And where do we plan on using it?

Reply:
Ok. Let's leave it out for now.

BTW, your comment says "to JSON" but the return type is `Dict`. Both of those can't be correct.

Reply:
Good point. I'd think serializing directly to JSON (return type of `str`) would be the best approach, as it would remove dealing with dictionaries and handling the associated typing problems.

Reply:
I'd be fine with that. If you just change the return type to `str` then we can resolve this convo.

Reply:
c644f93